lib/server-jobs.js

```diff
@@ -189,19 +192,43 @@ Job.prototype.failAsync = util.promisify(Job.prototype.fail);
 
 /********************************************************************/
 
-module.exports.init = function (callback) {
-  socketServer.io.on('connection', this.connection);
-  callback(null);
+let intervalId = null;
+
+module.exports.init = function () {
+  socketServer.io.on('connection', module.exports.connection);
+
+  // Start a periodic task to heartbeat all live jobs. We don't use a cronjob
+  // for this because we want this to run for every host.
+  intervalId = setInterval(() => {
+    const jobIds = Object.keys(module.exports.liveJobs);
+    if (jobIds.length === 0) return;
+
+    sqlDb.queryAsync(sql.update_heartbeats, { job_ids: jobIds }).catch((err) => {
+      logger.error('Error updating heartbeats for live server jobs', err);
+    });
+  }, 10 * 1000);
```

Comment:
Should this period be configurable?

Reply:
I think so -- it's easy to do and can't hurt?

Reply:
Done in 5346f32.